cmd/arktos-network-controller/app/controller.go

```diff
@@ -158,6 +158,33 @@ func (c *Controller) process(item interface{}) {
 	c.queue.Forget(item)
 }
 
+// manageNonFlatNetwork is the core logic to manage a non-flat typed network object
+func manageNonFlatNetwork(net *v1.Network, netClient arktos.Interface, svcClient kubernetes.Interface, toDeployDNS bool, domainName string) error {
+	var svc *corev1.Service
+	var err error
+	if len(net.Status.DNSServiceIP) == 0 {
+		if svc, err = createOrGetDNSService(net, svcClient); err != nil {
+			return fmt.Errorf("failed to get or create per-network DNS service: %v", err)
+		}
+	}
+
+	if toDeployDNS {
+		if err = deployDNSForNetwork(net, svcClient, domainName); err != nil {
+			return fmt.Errorf("failed to deploy per-network DNS: %v", err)
+		}
+	}
+
+	if len(svc.Spec.ClusterIP) > 0 {
```

Comment:
The DNS service is of type ClusterIP, so if len(svc.Spec.ClusterIP) > 0, we need to return an error.

Reply:
talked offline; will add a comment to explain the reason why cluster ip might be empty, to avoid confusion.